src/Security/Authorization/Core/src/AuthorizationFailureReason.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Encapsulates a reason why authorization failed.
+    /// </summary>
+    public class AuthorizationFailureReason
+    {
+        /// <summary>
+        /// Creates a new failure reason.
+        /// </summary>
+        /// <param name="handler">The handler responsible for this failure reason.</param>
+        /// <param name="message">The message describing the failure.</param>
+        public AuthorizationFailureReason(IAuthorizationHandler handler, string message)
+        {
+            Handler = handler;
+            Message = message;
+        }
+
+        /// <summary>
+        /// A message describing the failure reason.
+        /// </summary>
+        public string Message { get; set; }
```

Comment:
Does these properties needs a setter?
```suggestion
        public string Message { get; }
```

Reply:
Good point, they probably don't since they're set through the constructor.